docs-sphinx/user-guide/how-to-use-header-only-layoutbuilder.md

```diff
@@ -0,0 +1,260 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: C++14
+  language: C++14
+  name: xcpp14
+---
+
+How to use the header-only LayoutBuilder in C++
+===============================================
+
+What is header-only Layout Builder?
+-----------------------------------
+
+The header-only Layout Builder consists of a set of compile-time, templated, static C++ classes, implemented entirely in header file which can be dropped into any external project, and easily separable from the rest of the Awkward C++ codebase.
+
+The Layout Builder namespace consists of the following 14 types of Layout Builders:
+
+* `Numpy`
+* `ListOffset`
+* `List`
+* `Empty`
+* `EmptyRecord`
+* `Record`
+* `Tuple`
+* `Regular`
+* `Indexed`
+* `IndexedOption`
+* `Unmasked`
+* `ByteMasked`
+* `BitMasked`
+* `Union`
+
+All Builders except `Numpy` and `Empty` can take any other Builder as template parameters.
+These Builders are sufficient to build every type of Awkward Array.
+
+Why header-only Layout Builder?
+-------------------------------
+
+The users can directly include `LayoutBuilder.h` in their compilation, rather than linking against platform-specific libraries or worrying about native dependencies. This makes the integration of Awkward Arrays into other projects easier and more portable.
+
+The code is minimal, it does not include all of the code needed to use Awkward Arrays in Python, nor does it include references to Python or pybind11. The C++ users can use it to make arrays and then copy them to Python without any specialised data types - only raw buffers, strings, and integers.
+
+How to use Layout Builders?
+-----------------------------
+
+The following cpp-headers which reside in this [directory](https://github.com/scikit-hep/awkward/tree/main/src/awkward/_v2/cpp-headers/awkward) are needed to use Layout Builders.
+
+1. BuilderOptions.h
+2. GrowableBuffer.h
+3. LayoutBuilder.h
+4. utils.h
+
```

Comment:
I'd recommend to use an awkward installation (e.g. a versioned set of files  with pip install) and the config options described above. Otherwise, the users might copy a master branch at some point and we'd have difficulties tracing the bugs reports.